storm-core/src/clj/org/apache/storm/zookeeper.clj

```diff
@@ -57,15 +46,15 @@
    :auth-conf nil]
   (let [fk (Utils/newCurator conf servers port root (when auth-conf (ZookeeperAuthInfo. auth-conf)))]
     (.. fk
-        (getCuratorListenable)
-        (addListener
-          (reify CuratorListener
-            (^void eventReceived [this ^CuratorFramework _fk ^CuratorEvent e]
-                   (when (= (.getType e) CuratorEventType/WATCHED)
-                     (let [^WatchedEvent event (.getWatchedEvent e)]
-                       (watcher (zk-keeper-states (.getState event))
-                                (zk-event-types (.getType event))
-                                (.getPath event))))))))
+      (getCuratorListenable)
+      (addListener
+        (reify CuratorListener
+          (^void eventReceived [this ^CuratorFramework _fk ^CuratorEvent e]
+            (when (= (.getType e) CuratorEventType/WATCHED)
+              (let [^WatchedEvent event (.getWatchedEvent e)]
+                (watcher (ZkKeeperStates/getStateName (.getState event))
+                  (ZkEventTypes/getTypeName (.getType event))
```

Comment:
These are returning Strings, not symbols. We probably want to just return the enum itself.  Also anyone who is using those symbols needs to be updated.  When I run the tests I am getting NPEs because of this.

It looks like you will mostly need to update 

./storm-core/src/clj/org/apache/storm/cluster_state/zookeeper_state_factory.clj

and

./storm-core/test/clj/org/apache/storm/cluster_test.clj